include/circt/Dialect/HW/HWAggregates.td

```diff
@@ -122,7 +122,6 @@ class IndexBitWidthConstraint<string index, string input>
                    input, index,
                    [{
     IntegerType::get($_self.getContext(),
-                     type_cast<ArrayType>($_self).getSize() == 1 ? 1 :
```

Comment:
I would prefer to keep the special case and just allow i0 as well, so we don't change too much at once.